### DISCUSSION TITLE
Appeal for an open discussion culture in openPDF

### DISCUSSION THREAD

**Discussion by Lonzak** [LibrePDF/OpenPDF#1053] [General]
As a dedicated member of this project, I deeply value the opportunity to contribute through our collective efforts and the exchange of knowledge and ideas. The cornerstones of OpenSource – openness, collaboration, and transparency – are values that I hold dear. Until recently I thought this project has a similar understanding. At least in the [description ](https://github.com/LibrePDF/OpenPDF/discussions/761) one can read "place to connect with other members of our community", "ask questions", "share ideas", "engage with other community members", "open-minded" and more...

Recently, however, I have noticed that my attempts to ask questions or raise issues in the discussion area of this project are immediately met with obstacles. My entries are consistently locked or deleted before any discussion can take place. This practice not only prevents me and others from making constructive contributions but also deprives the entire community of the chance to participate and collectively find solutions.

I understand that moderation is necessary to keep our discussion forums productive and focused. Yet, I believe that a culture where locking or deleting posts is the first response, does not align with the principles of OpenSource. It creates an atmosphere of restriction that is neither conducive to innovation nor beneficial to the community.

Therefore, I appeal to reconsider our discussion culture. Let us find a way that allows for effective moderation while also fostering open, respectful, and constructive discussions. A culture where all voices are heard and where we can grow together is crucial not only for the development of this project but also for the well-being of each one of us.

I am convinced that we can find solutions that preserve the integrity of our project and promote a vibrant and inclusive discussion culture. 

Lonzak

PS: 
I raised issues (#1033 #1034) and they were closed and locked immediately. Ok, maybe the section was wrong. But then I opened a discussion (#1052) and answered another discussion (#1037) and both were quickly locked/deleted as well. One answer was "Contact asturio if you want to help maintain support for older Java versions." But the discussion was locked immediately, so I couldn't even answer. I don't know how to contact @asturio. He could have replied to the discussion - I mean that is what the discussion section is for, isn't it?)

**Comment by asturio** [comment]
Hi @Lonzak ,

thank you for pointing it out. When I opened the Discussion Space in GitHub for OpenPDF I wanted to solve some problems:

- I wanted to avoid having random "Issues" raised from users, who just need some help using the library. I think a better place for questions was the "discussion"s area. As a forum one can search for the solution there and if nothing is found ask a new question.
- As I don't have too much time, I was also interested in a place where other users can help each other.

I see the discussion area as a "private" "Stack overflow" for OpenPDF. An alternative would be shift all the discussions to Stack-overflow. To be sincere, I have never paid much attention to this area, hopping other users will help out each other.

I also have an e-mail filter configured, so all e-mails from GitHub OpenPDF goes in it's own folder, and this folder I don't look every day :-( .

@andreasrosdal is helping out again since some time releasing OpenPDF a lot more than I did, and that is great. A lot of PRs were merged in the latest time and some issues were also closed.

I'll take a look at the discussions you pointed out.

**Reply by asturio** [reply]
BTW: my e-mail address can the found in any of my commits. That is the best way to contact me directly.

**Comment by asturio** [comment]
I created a new branch 1.3-java8 from head, so it is not yet Java8 compatible. I'll will try to downgrade it to Java 8, and create a new 1.3-Release from there. @Lonzak if you want to help doing that, feel free in creating some PRs.

**Comment by Lonzak** [comment]
I did that already - and had to do 3 changes:

1.)
```
SimplePatternParser  Line 100 - 
try (stream) => try{
```

2.)
openpdf/pom.xml:
```
<artifactId>mockito-core</artifactId>
<version>5.10.0</version>  => <version>4.11.0</version>

```
3.)
parent pom.xml
All Java11/17 version to jdk8

=> sure I'll create a PR

**Comment by asturio** [comment]
I just realized that master is already at Java 17 (2.0). And there was also a 1.4-Version of OpenPDF. So what about this?

- 1.3: Java 8
- 1.4 Java 11 (branch don't exist now, I would create one from 1.3 (all are kind of from head today)
- master: Java 17

Would that be ok? And 1.3 and 1.4 would only receive bugfixes. @andreasrosdal what do you think?

**Comment by Lonzak** [comment]
Yes that would be a good solution!

What about the 1.3.X version which have been compiled and release with Java8? I think we should re-release those...

**Comment by Lonzak** [comment]
Done: https://github.com/LibrePDF/OpenPDF/pull/1056

**Reply by xexes** [reply]
MappedRandomAccessFile.java  requires undoing changes

**Reply by Lonzak** [reply]
For me it is working with Java8 compiler. What do you think needs to be undone?

**Reply by xexes** [reply]
old cleaner  metods   removed in https://github.com/LibrePDF/OpenPDF/commit/5a81749f0b7cee186ac13f44afea9929cd65f160  for java 8

**Reply by asturio** [reply]
Any PR in that aspect could also be added. Feel free in creating one.

**Comment by asturio** [comment]
So, I thing the branch is now ok.

Older releases of 1.3 won't be re-released, but I can release the next 1.3 now, which "fixes" the "not compatible to Java 8 anymore". Making also a 1.4 so compatible again with Java 11. But development will be made further only in master (which is now Java 21!?!)

**Comment by Lonzak** [comment]
Above you have written:

"master: Java 17"

21 would be quite a jump (again)....or?

**Reply by asturio** [reply]
Just to clarify what is in master: There the java-release is configured for Java 17. Only in the maven builds we build if using Java 17 and 21.

**Comment by andreasrosdalw** [comment]
I would recommend to keep OpenPDF 2.x at Java 17, and further recommend to keep the API stable for a long time now. It has already been released for Java 17+. For example the Spring framework requires Java 17 in the latest version now.

**Comment by asturio** [comment]
I think the way seems cleared now.

I would also keep 2.0 at Java 17 (and changing to Java 21 later on a new Major-Version (3.0?)).

With the "new" release cycle for Java a lot of libraries are struggling to keep up.

I think it is important to keep this compatibility with older versions for some time, as one can not always switch easily to the next java version. Not because of OpenPDF, but because of other dependencies a project can have.

So sometimes you need to wait for these dependencies be updated to a newer java version, and sometimes you need to change that dependency altogether.
